### PR TITLE
Upgrade web server for OAuth connection process [add-on: google_assistant]

### PR DESCRIPTION
**Pull request by ericmatte**
Hi,

I really like the Google Assistant SDK addon! It makes it easier to implement it alongside Home-Assistant.

Description
-----------

Currently, this addon is in "Experimental" stage and the web server to authenticate looks like this:

Example 1

And after entering the token, I am welcomed with an error 500, just before the server shuts down (even thought the process has successfully completed and Google Assistant is now properly running on my device):

Example 1

---

I thought I could offer my help and give a more finished look to the credentials process web server.

Proposed change
------------------

This is what I propose: a more Home-Assistant like native look:

Screen Shot 2020-05-03 at 21 05 00

And after entering the token/code:

Screen Shot 2020-05-03 at 21 05 27

What changed
--------------

- Added a css file for styling, logo image and favicon
- In the UI, say "code" instead of "token", to align with how Google name that string
_After authenticating on the Google side:_ Example 1
- Prevent error 500 after entering the token/code.
    - The problem was that when calling `sys.exit(0)` without returning any response to the browser, the page `/token` is requested twice. The second time, the token becomes invalid which raises the exception.
    - Now we show a simple success message, and gracefully quit cherrypy from another thread.
- If the user requests the page `/token?token=a_token` with an invalid token, he will be redirected to the home page to restart the authentication process, and the error will be shown in the logs.

I think that this new UI would give a great look and a more frictionless authentication process.
Let me know what you think.

This is potentially a fix for https://github.com/home-assistant/hassio-addons/issues/1254.

### CODE REVIEW

**Comment by ericmatte** [comment]
> Looks good, thanks. Can we offload the static folder to `/usr/share/xxx`? Inside S6 folder is not a good place.

Sure

**Comment by ericmatte** [comment]
@pvizeli I've made the [change](https://github.com/home-assistant/hassio-addons/pull/1267/commits/88f536447dfdc743b0449dac0d467ea299848b48). Let me know what you think!

**Comment by ericmatte** [comment]
Done

File: google_assistant/config.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "Google Assistant SDK",
-  "version": "2.3.2",
+  "version": "2.3.3",
```
Comment (pvizeli) [inline google_assistant/config.json:3]:
```suggestion
  "version": "2.4.0",
```

File: google_assistant/CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,10 @@
 # Changelog
 
+## 2.3.3
```
Comment (pvizeli) [inline google_assistant/CHANGELOG.md:3]:
```suggestion
## 2.4.0
```

**Comment by pvizeli** [comment]
thanks

**Comment by ericmatte** [comment]
@pvizeli  I tried to install the new version of the add-on on my raspberry pi, and I got this error:
```
20-05-06 12:54:21 ERROR (SyncWorker_19) [supervisor.docker.interface] Can't install homeassistant/armv7-addon-google_assistant:2.4.0 -> 404 Client Error: Not Found ("manifest for homeassistant/armv7-addon-google_assistant:2.4.0 not found: manifest unknown: manifest unknown").
```

Do you need to push a new version of `homeassistant/armv7-addon-google_assistant:2.4.0` to your Docker hub repository^

**Comment by ericmatte** [comment]
I've just retried installing it and it worked!

```diff
+ 20-05-06 15:16:59 INFO (SyncWorker_5) [supervisor.docker.interface] Pull image homeassistant/armv7-addon-google_assistant tag 2.4.0.
+ 20-05-06 15:25:46 INFO (MainThread) [supervisor.addons] Add-on 'core_google_assistant' successfully installed
+ 20-05-06 15:26:32 INFO (SyncWorker_19) [supervisor.docker.addon] Start Docker add-on homeassistant/armv7-addon-google_assistant with version 2.4.0
```

Probably that we need to wait for the image to be built on the Docker repository before it's accessible.

It would be a good idea to not show add-on updates while their images are being built, has this could confuse users into thinking it's broken.

**Pull request merged**